api/scpca_portal/validators.py

```diff
@@ -54,3 +56,89 @@ def validate_project_ids(cls, instance):
                 # TODO: add custom exception
                 raise ValueError(f"Invalid project ID format: {project_id}")
         return instance
+
+
+class DatasetDataResourceExistence:
```

Comment:
prefixes:
- invalid (not in db)
- existing (in db / confirmed)
- data (from input)

You can change these but having consistent names for the context would be helpful here.